src/styles/blocks/header.scss

```diff
@@ -38,5 +36,11 @@
     text-transform: uppercase;
     font-size: 12px;
     font-weight: 500;
+    transition: color 300ms;
+  }
+
+  &__link:hover {
+    transition: color 300ms;
```

Comment:
The transition should work in both ways (not only for hover)

https://user-images.githubusercontent.com/99403198/229280840-5125279d-d557-40ee-8a7e-0192179d8edf.mov